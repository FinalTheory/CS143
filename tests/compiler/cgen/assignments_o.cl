-- Test of assignment, and some recursion

class Main inherits IO {

   foo(x:Int) : Int {
		if x < 10 then {
	    	x <- x+foo(x+1);
			x + 1;
		}
		else
			1
		fi
   };

   x : Int <- 0;
   main() : Object {{
     x <- foo(4);
	 out_int(x);
     self;
   }};
};
   
