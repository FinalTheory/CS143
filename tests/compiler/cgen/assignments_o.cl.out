SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
46COOL program successfully executed
Stats -- #instructions : 1480
         #reads : 333  #writes 209  #branches 322  #other 616
